packages/ui-components/src/components/tooltip/tooltip.tsx

```diff
@@ -55,12 +57,23 @@ export class KvTooltip implements ITooltip {
 	};
 
 	private showTooltipHandler = () => {
-		if (this.delay) {
-			this.timeoutDelayId = window.setTimeout(() => {
-				this.showTooltip();
-			}, this.delay);
-		} else {
-			this.showTooltip();
+		const range = new Range();
+		range.selectNodeContents(this.el);
+
+		switch (this.behaviour) {
```

Comment:
Since all the cases can use the same resource code, I suggest to change this in a more readable way:

```
if (this.behaviour === ETooltipBehaviour.Truncate) {
	const { width: rectWidth } = range.getBoundingClientRect();
	const { width: elWidth } = this.el.getBoundingClientRect();

	if (rectWidth < elWidth) return;
}

if (this.delay) {
	this.timeoutDelayId = window.setTimeout(() => {
		this.showTooltip();
	}, this.delay);
} else {
	this.showTooltip();
}
```